src/charm.py

```diff
@@ -154,6 +159,21 @@ def tls_available(self) -> bool:
             and (self.cert_handler.ca_cert is not None)
         )
 
+    @property
+    def _s3_config(self) -> Optional[dict]:
```

Comment:
maybe also leave a comment or a TODO here explaining why we went with a dict instead of the pydantic types

Reply:
yeh, don't like that either. Will spend some time to type it all properly